src/svm/solanaProgramUtils.ts

```diff
@@ -189,3 +191,69 @@ export async function subscribeToCpiEventsForProgram(
 
   return subscriptionId;
 }
+
+/**
+ * Class for DepositId.
+ */
+class DepositId {
+  value: Uint8Array; // Fixed-length array as Uint8Array
+
+  constructor(properties: { value: Uint8Array }) {
+    this.value = properties.value;
+  }
+}
+
+/**
+ * Borsh schema for deserializing DepositId.
+ */
+const depositIdSchema = new Map(
+  [[DepositId, { kind: "struct", fields: [["value", [32]]] }]] // Fixed array [u8; 32]
+);
+
+/**
+ * Parses depositId: checks if only the first 4 bytes are non-zero and returns a u32 or deserializes the full array.
+ */
+function parseDepositId(value: Uint8Array): string {
+  const restAreZero = value.slice(4).every((byte) => byte === 0);
+
+  if (restAreZero) {
+    // Parse the first 4 bytes as a little-endian u32
+    const u32Value = new DataView(value.buffer).getUint32(0, true); // true for little-endian
+    return u32Value.toString();
+  }
+
+  // Deserialize the full depositId using the Borsh schema
+  const depositId = deserialize(depositIdSchema, DepositId, Buffer.from(value));
+  return new BN(depositId.value).toString();
+}
+
+/**
+ * Stringifies a CPI event.
+ */
+export function stringifyCpiEvent(obj: any): any {
+  if (obj?.constructor?.toString()?.includes("PublicKey")) {
+    if (obj.toString().includes("111111111111")) {
```

Comment:
unlikely to have zeroes anywhere else than in the beginning, though, still better to restrict that only the first 12 bytes matching 0 are translated as EVM addresses

Reply:
100%